src/mds/MDCache.h

```diff
@@ -1206,7 +1206,7 @@ class MDCache {
   std::set<CDir*>   rejoin_undef_dirfrags;
   std::map<mds_rank_t, std::set<CInode*> > rejoin_unlinked_inodes;
 
-  std::vector<CInode*> rejoin_recover_q, rejoin_check_q;
+  std::set<CInode*> rejoin_recover_q, rejoin_check_q;
```

Comment:
How about using a flat_set instead?

Reply:
Since the MDS won't do random access to these two queues, IMO it doesn't make much sense to use `flat_set` (?).

Reply:
Right. There is iteration, but called only once.